Author: Adam Majer <adamm@zombino.com>
Description: Use "const char *" instead of "char *" to prevent warnings with g++ (see bug #502581)
Forwarded: no
diff -ru cunit-2.1-0.dfsg.old/CUnit/Headers/TestRun.h cunit-2.1-0.dfsg/CUnit/Headers/TestRun.h
--- cunit-2.1-0.dfsg.old/CUnit/Headers/TestRun.h      2006-01-29 21:48:26.000000000 -0600
+++ cunit-2.1-0.dfsg/CUnit/Headers/TestRun.h	      2008-10-17 23:03:14.665086520 -0500
@@ -172,9 +172,9 @@
 /* Assertion implementation function. */
 CU_EXPORT CU_BOOL CU_assertImplementation(CU_BOOL bValue,
                                           unsigned int uiLine,
-                                          char strCondition[],
-                                          char strFile[],
-                                          char strFunction[],
+                                          const char strCondition[],
+                                          const char strFile[],
+                                          const char strFunction[],
                                           CU_BOOL bFatal);
 
 #ifdef USE_DEPRECATED_CUNIT_NAMES
diff -ru cunit-2.1-0.dfsg.old/CUnit/Sources/Framework/TestRun.c cunit-2.1-0.dfsg/CUnit/Sources/Framework/TestRun.c
--- cunit-2.1-0.dfsg.old/CUnit/Sources/Framework/TestRun.c	2006-01-29 21:52:34.000000000 -0600
+++ cunit-2.1-0.dfsg/CUnit/Sources/Framework/TestRun.c		2008-10-17 23:05:43.191046200 -0500
@@ -87,8 +87,8 @@
 static void         add_failure(CU_pFailureRecord* ppFailure,
                                 CU_pRunSummary pRunSummary,
                                 unsigned int uiLineNumber,
-                                char szCondition[],
-                                char szFileName[],
+                                const char szCondition[],
+                                const char szFileName[],
                                 CU_pSuite pSuite,
                                 CU_pTest pTest);
 
@@ -119,9 +119,9 @@
  */
 CU_BOOL CU_assertImplementation(CU_BOOL bValue, 
                                 unsigned int uiLine,
-                                char strCondition[], 
-                                char strFile[],
-                                char strFunction[], 
+                                const char strCondition[], 
+                                const char strFile[],
+                                const char strFunction[], 
                                 CU_BOOL bFatal)
 {
   /* not used in current implementation - stop compiler warning */
@@ -587,8 +587,8 @@
 void add_failure(CU_pFailureRecord* ppFailure,
                  CU_pRunSummary pRunSummary,
                  unsigned int uiLineNumber,
-                 char szCondition[],
-                 char szFileName[],
+                 const char szCondition[],
+                 const char szFileName[],
                  CU_pSuite pSuite,
                  CU_pTest pTest)
 {
